src/core/options-updater.cpp

```diff
@@ -0,0 +1,77 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+
+
+#include "options-updater.h"
+#include <proc/synthetic-stream.h>
+
+
+namespace librealsense {
+
+
+options_updater::options_updater( std::chrono::milliseconds update_interval )
+    : _update_interval( update_interval ),
+    _updater([this](dispatcher::cancellable_timer cancellable_timer) { update_loop(cancellable_timer); })
+{
+}
+
+options_updater::~options_updater()
+{
+    _updater.stop();
+}
+
+void options_updater::register_option( rs2_option id, std::shared_ptr< option > option )
+{
+    _options[id] = option;
+    _option_values[id] = option->query();
+
+    if( ! _updater.is_active() )
+        _updater.start();
```

Comment:
Don't we want to start on first subscriber?

Reply:
in 'subscribe_to_updates'?

Reply:
It's kind of a mix, we want to work if we have both registered options and users to notify. I will fix the logic.

Reply:
Done